x-pack/legacy/plugins/siem/public/store/timeline/model.ts

```diff
@@ -5,17 +5,36 @@
  */
 
 import { Filter } from '../../../../../../../src/plugins/data/public';
-import { ColumnHeader } from '../../components/timeline/body/column_headers/column_header';
 import { DataProvider } from '../../components/timeline/data_providers/data_provider';
-import { DEFAULT_TIMELINE_WIDTH } from '../../components/timeline/body/helpers';
-import { defaultHeaders } from '../../components/timeline/body/column_headers/default_headers';
 import { Sort } from '../../components/timeline/body/sort';
-import { Direction, PinnedEvent, TimelineNonEcsData } from '../../graphql/types';
+import { PinnedEvent, TimelineNonEcsData } from '../../graphql/types';
 import { KueryFilterQuery, SerializedFilterQuery } from '../model';
 
 export const DEFAULT_PAGE_COUNT = 2; // Eui Pager will not render unless this is a minimum of 2 pages
 export type KqlMode = 'filter' | 'search';
 export type EventType = 'all' | 'raw' | 'signal';
+
+export type ColumnHeaderType = 'not-filtered' | 'text-filter';
+
+/** Uniquely identifies a column */
+export type ColumnId = string;
+
+/** The specification of a column header */
+export interface ColumnHeader {
```

Comment:
Would this be more accurate as `ColumnHeaderOptions`? I personally found it a little confusing that a `ColumnHeader` component receives a `ColumnHeader` prop.